src/Weeder/Run.hs

```diff
@@ -173,3 +176,17 @@ runWeeder weederConfig@Config{ rootPatterns, typeClassRoots, rootInstances, root
 displayDeclaration :: Declaration -> String
 displayDeclaration d = 
   moduleNameString ( moduleName ( declModule d ) ) <> "." <> occNameString ( declOccName d )
+
+-- | This computes the same as (Set.\\) but is faster by assuming that the set difference will be small.
+-- I.e. there will be more non-weeds than weeds.
+--
+fastSpecialSetDifference :: Set Declaration -> Set Declaration -> Set Declaration
+fastSpecialSetDifference allDecls usedDecls = 
+  let bloom = BloomFilter.easyList 0.05 (Set.toList usedDecls)
+      -- The elem docs say:
+      -- @
+      -- If the value is present, return True. If the value is not present, there is still some possibility that True will be returned.
+      -- @
+      -- I.e. if some declaration is a weed, it will definitely show up in the result, but also some weeds will show up in the result.
+      -- So we need to do another set difference afterwards, but with a much smaller set.
+  in Set.difference (Set.filter (not . (`BloomFilter.elem` bloom)) allDecls) usedDecls
```

Comment:
I'm about 50% sure I got this backward in some way, so that might be part of the issue as well.